sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -401,6 +401,13 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     }
   }
 
+  override def alterTableStats(tableDefinition: CatalogTable): Unit = withClient {
```

Comment:
@viirya There's two ways to set/replace properties:
1. We use statistics info in CatalogTable to set these properties - this is called by analyze command and I put this logic into alterTableStats method;
2. We set these properties directly - this is the path of alter table command.
If we put alterTableStats logic in the original alterTable method, we can't set the properties by the second way, because they will be always replaced by stats in CatalogTable.